src/lib/wsman-filter.c

```diff
@@ -357,7 +357,8 @@ filter_t * filter_deserialize(WsXmlNodeH node, const char *ns)
 {
 	char *dialect = NULL;
 	int properNum = 0;
-	int i = 0;
+	unsigned int i = 0;
+	int j = 0;
```

Comment:
Then this line can go to the `if(strcmp(filter->dialect , WSM_ASSOCIATION_FILTER_DIALECT) == 0) {` block